mlflow/models/evaluation/base.py

```diff
@@ -957,6 +951,26 @@ def _validate(validation_thresholds, candidate_metrics, baseline_metrics=None):
     raise ModelValidationFailedException(message=os.linesep.join(failure_messages))
 
 
+def _convert_data_to_mlflow_dataset(data, targets=None, dataset_path=None):
+    """Convert input data to mlflow dataset."""
+    if "pyspark" in sys.modules:
+        from pyspark.sql import DataFrame as SparkDataFrame
+
+    if isinstance(data, Dataset):
+        return data
```

Comment:
can we pull this check into `_evaluate()` rather than in `_convert_data_to_mlflow_dataset()`? i think it is easier to reason about only calling `_convert_data_to_mlflow_dataset()` if we actually need to convert something.

Reply:
This is actually related to a comment below on if we should throw a `TypeError` if the type does not fall into our allowed categories. 

My understanding is that we want to convert data of every legal format into a Dataset for handling, so this `_convert_data_to_mlflow_dataset()` is also a sanity check on input data type, which is why every allowed data type is shown here. `list` is missing here, I will add that.

Reply:
Moved the if check around since we are dropping the `TypeError` in this PR.